### ISSUE TITLE
Tabsets don't work in pandoc 2.8

### ISSUE THREAD

**Issue by mnazarov** [rstudio/rmarkdown#1723]
Another compatibility issue with pandoc 2.8+ (cf. https://github.com/rstudio/bookdown/issues/832) happens when using `.tabset`s. The difference is due to the following change: when running
```
pandoc -t html4 --section-divs <<< "# test {.tabset}"
```
gives in pandoc 2.7.3:
```
<div id="test" class="section level1 tabset">
<h1>test</h1>
</div>
```
and in pandoc 2.8:
```
<div id="test" class="section level1">
<h1 class="tabset">test</h1>
</div>
```
so the selector https://github.com/rstudio/rmarkdown/blob/07a0ca4c3b8a11cbaf3c8050fe2064b91de2f0b8/inst/rmd/h/navigation-1.1/tabsets.js#L136 fails. A solution is to change the above to
```js
var tabsets = $("div.section.tabset");
if (tabsets.length == 0)
  tabsets = $("div.section").has(">.tabset");
```
also https://github.com/rstudio/rmarkdown/blob/07a0ca4c3b8a11cbaf3c8050fe2064b91de2f0b8/inst/rmd/h/navigation-1.1/tabsets.js#L61 needs to be changed to 
```
var match = tabset.attr('class').match(/level(\d)+/);
```
but it is a bit more tricky with sticky tabsets, hopefully someone can take it up from here...

---
<details>
  <summary>minimal example:</summary>

```
---
title: test tabsets
output: html_document
---

# Tabs {.tabset}

## Number 1

Text 1

## Numer 2

Text 2
```
</details>

---

<details>
  <summary>session info</summary>

```
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Locale:
  LC_CTYPE=en_US.UTF-8          LC_NUMERIC=C                  LC_TIME=en_IE.UTF-8           LC_COLLATE=en_US.UTF-8       
  LC_MONETARY=en_IE.UTF-8       LC_MESSAGES=en_US.UTF-8       LC_PAPER=en_IE.UTF-8          LC_NAME=en_IE.UTF-8          
  LC_ADDRESS=en_IE.UTF-8        LC_TELEPHONE=en_IE.UTF-8      LC_MEASUREMENT=en_IE.UTF-8    LC_IDENTIFICATION=en_IE.UTF-8

Package version:
  base64enc_0.1.3  digest_0.6.21    evaluate_0.14    glue_1.3.1       graphics_3.6.1   grDevices_3.6.1  highr_0.8        htmltools_0.4.0 
  jsonlite_1.6     knitr_1.25       magrittr_1.5     markdown_1.1     methods_3.6.1    mime_0.7         Rcpp_1.0.2       rlang_0.4.0     
  rmarkdown_1.18.5 stats_3.6.1      stringi_1.4.3    stringr_1.4.0    tinytex_0.16     tools_3.6.1      utils_3.6.1      xfun_0.10       
  yaml_2.2.0      

Pandoc version: 2.8
```
</details>

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by yihui** [comment]
Thanks for the careful investigation and helpful hints! I'll take a look as soon as possible.

**Comment by yihui** [comment]
I reported the problem to https://github.com/jgm/pandoc/issues/5965 and it was fixed in Pandoc. Tabsets will work with the next version of Pandoc again.

I have also pushed a fix specifically for Pandoc 2.8 in our `tabsets.js`. Thanks for the report!

**Comment by cderv** [comment]
FWIW, pandoc 2.9 has been release and include a fix to this issue I think.
https://github.com/jgm/pandoc/releases/tag/2.9

**Comment by yihui** [comment]
jgm/pandoc#5965 has been (partially) resolved, and Pandoc 2.9 has been released 20 days after 2.8. I recommend that you upgrade to 2.9, because this issue will appear again in the next release of **rmarkdown** if you continue to use Pandoc 2.8.
